packages/flutter/lib/src/rendering/error.dart

```diff
@@ -50,6 +51,9 @@ class RenderErrorBox extends RenderBox {
 
   /// The message to attempt to display at paint time.
   final String message;
+  final double systemTopPadding = Zone.current['systemTopPadding'] == null
```

Comment:
I mean, fundamentally I would limit the number of places interacting with non-mockable global components like dart:ui so that this just doesn't really become a problem. Would you move all of dart:ui into a Zone?

I find the use of zones for this kind of thing to be very aesthetically unsatisfactory from an API perspective. Also, in addition to the loss of strong typing (by which I specifically mean that the analyzer can't verify that code reading out of zones in this way is correct), you lose dartdocs and autocompletion.

> Another benefit of the Zone approach is that it will let you test code that someone else wrote that wasn't expecting to be tested in this way.

Not really, because someone who isn't expecting to be testing in this way wouldn't use zones (unless you mean specifically for things we are forcing them to use zones for, but then it seems like we're the ones who are expecting things to be tested this way, if you see what I mean).

The flutter_tools code uses zones a bit for this. I don't think in practice that it's been a win over more straight-forward injection.